### PR TITLE
Added objects,custom settings,page layouts

### PR DESCRIPTION
**Pull request by csankit0108**
1. Added Case Comment,Case Collaborator object, Email Alert Configuration custom setting
2. Fixed test classes & naming conventions
3. Updated package.xml to include only relevant types.

### CODE REVIEW

File: Delmar/force-app/main/default/classes/DEL_EmailMessagesTriggerHandler.cls [hunk 1]
```diff
@@ -22,12 +22,12 @@ public with sharing class DEL_EmailMessagesTriggerHandler {
         Pattern pattern = Pattern.compile(strPattern);
```
Comment (pradeepchary) [inline Delmar/force-app/main/default/classes/DEL_EmailMessagesTriggerHandler.cls:22]: what is this pattern?

File: Delmar/force-app/main/default/classes/DEL_EmailMessagesTriggerHandler.cls [hunk 2]
```diff
@@ -22,12 +22,12 @@ public with sharing class DEL_EmailMessagesTriggerHandler {
...
         for (EmailMessage objEmailMessage : list_EmailMessages) {
-            if ((objEmailMessage.ThreadIdentifier <> objEmailMessage.MessageIdentifier) && objEmailMessage.Incoming){
+            if ((objEmailMessage.ThreadIdentifier <> objEmailMessage.MessageIdentifier) && objEmailMessage.Incoming) {
                 if (String.isNotBlank(objEmailMessage.ParentId)) {
                     DEL_CaseComment__c objCaseComment = new DEL_CaseComment__c();
                     objCaseComment.Case__c = objEmailMessage.ParentId;
                     if (String.isNotBlank(objEmailMessage.HtmlBody)) {
-                        String[] list_SplitResults = pattern.split(objEmailMessage.HtmlBody);
+                        List<String> list_SplitResults = pattern.split(objEmailMessage.HtmlBody);
                         objCaseComment.Body__c = list_SplitResults.get(0).stripHtmlTags();
```
Comment (pradeepchary) [inline Delmar/force-app/main/default/classes/DEL_EmailMessagesTriggerHandler.cls:31]: Add a IF condition checking the list is not empty before hard coding the 0 index element

File: Delmar/force-app/main/default/layouts/DEL_CaseCollaborator__c-Case Collaborator Layout.layout-meta.xml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        </layoutColumns>
+        <layoutColumns>
+            <layoutItems>
+                <behavior>Readonly</behavior>
+                <field>LastModifiedById</field>
+            </layoutItems>
+        </layoutColumns>
+        <style>TwoColumnsTopToBottom</style>
+    </layoutSections>
+    <layoutSections>
```
Comment (pradeepchary) [inline Delmar/force-app/main/default/layouts/DEL_CaseCollaborator__c-Case Collaborator Layout.layout-meta.xml:50]: There is nothing in this layout section, please remove

File: Delmar/force-app/main/default/classes/DEL_EmailMessagesTriggerHandler.cls (hunk 1 above)
Comment (csankit0108) [reply Delmar/force-app/main/default/classes/DEL_EmailMessagesTriggerHandler.cls:22]: @pradeepchary Rakesh is using this to get the latest message from the email thread

File: Delmar/force-app/main/default/layouts/DEL_CaseComment__c-Case Comment Layout.layout-meta.xml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        </layoutColumns>
+        <layoutColumns>
+            <layoutItems>
+                <behavior>Readonly</behavior>
+                <field>LastModifiedById</field>
+            </layoutItems>
+        </layoutColumns>
+        <style>TwoColumnsTopToBottom</style>
+    </layoutSections>
+    <layoutSections>
```
Comment (pradeepchary) [inline Delmar/force-app/main/default/layouts/DEL_CaseComment__c-Case Comment Layout.layout-meta.xml:50]: remove the custom links layout section

File: Delmar/force-app/main/default/objects/DEL_CaseComment__c/fields/CommentCreatedName__c.field-meta.xml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fullName>CommentCreatedName__c</fullName>
+    <externalId>false</externalId>
+    <label>Comment Created Name</label>
```
Comment (pradeepchary) [inline Delmar/force-app/main/default/objects/DEL_CaseComment__c/fields/CommentCreatedName__c.field-meta.xml:5]:
```suggestion
    <label>Comment Created By Name</label>
```

File: Delmar/force-app/main/default/objects/DEL_EmailAlertConfiguration__c/DEL_EmailAlertConfiguration__c.object-meta.xml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (pradeepchary) [inline Delmar/force-app/main/default/objects/DEL_EmailAlertConfiguration__c/DEL_EmailAlertConfiguration__c.object-meta.xml:1]: please add description, what is this object about

File: Delmar/force-app/main/default/objects/DEL_EmailAlertConfiguration__c/fields/CaseCommentEmailTemplate__c.field-meta.xml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
```
Comment (pradeepchary) [inline Delmar/force-app/main/default/objects/DEL_EmailAlertConfiguration__c/fields/CaseCommentEmailTemplate__c.field-meta.xml:2]: please add description/help text, what is this field about

File: Delmar/force-app/main/default/objects/DEL_EmailAlertConfiguration__c/fields/CaseCommentFields__c.field-meta.xml [hunk 8]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fullName>CaseCommentFields__c</fullName>
```
Comment (pradeepchary) [inline Delmar/force-app/main/default/objects/DEL_EmailAlertConfiguration__c/fields/CaseCommentFields__c.field-meta.xml:3]: please add description/help text, what is this field about

File: Delmar/force-app/main/default/objects/DEL_EmailAlertConfiguration__c/fields/CaseQueryFields__c.field-meta.xml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
```
Comment (pradeepchary) [inline Delmar/force-app/main/default/objects/DEL_EmailAlertConfiguration__c/fields/CaseQueryFields__c.field-meta.xml:2]: please add description/help text, what is this field about

File: Delmar/force-app/main/default/objects/DEL_EmailAlertConfiguration__c/fields/MergeFields__c.field-meta.xml [hunk 10]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
```
Comment (pradeepchary) [inline Delmar/force-app/main/default/objects/DEL_EmailAlertConfiguration__c/fields/MergeFields__c.field-meta.xml:2]: please add description/help text, what is this field about

File: Delmar/force-app/main/default/objects/DEL_TriggerConfiguration__mdt/DEL_TriggerConfiguration__mdt.object-meta.xml [hunk 11]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <CustomObject xmlns="http://soap.sforce.com/2006/04/metadata">
-    <label>DEL_TriggerConfig</label>
-    <pluralLabel>DEL_TriggerConfigs</pluralLabel>
+    <label>DEL_TriggerConfiguration</label>
```
Comment (pradeepchary) [inline Delmar/force-app/main/default/objects/DEL_TriggerConfiguration__mdt/DEL_TriggerConfiguration__mdt.object-meta.xml:3]: API should be DEL_TriggerConfiguration
Label - Trigger Configuration
Plural Label - Trigger Configurations

File: Delmar/force-app/main/default/triggers/DEL_EmailMessagesTrigger.trigger [hunk 12]
```diff
@@ -1,7 +1,7 @@
 trigger DEL_EmailMessagesTrigger on EmailMessage (after insert) {
     if (
-        (DEL_TriggerConfig__mdt.getInstance('CMDEL0001') != null &&
-        !DEL_TriggerConfig__mdt.getInstance('CMDEL0001').IsActive__c) ||
+        (DEL_TriggerConfiguration__mdt.getInstance('CMDEL0001') != null &&
```
Comment (pradeepchary) [inline Delmar/force-app/main/default/triggers/DEL_EmailMessagesTrigger.trigger:3]: Please add comments describing what is this if condition about and what is CMDEL0001

File: Delmar/manifest/package.xml [hunk 13]
```diff
@@ -1,31 +1,32 @@
...
     </types>
     <types>
-        <members>DEL_TriggerConfig__mdt-DEL_TriggerConfig Layout</members>
+        <members>DEL_CaseCollaborator__c-Case Collaborator Layout</members>
+        <members>DEL_TriggerConfiguration__mdt-DEL_TriggerConfiguration Layout</members>
         <name>Layout</name>
     </types>
     <types>
-        <members>*</members>
+        <members>del_commentsPage</members>
```
Comment (pradeepchary) [inline Delmar/manifest/package.xml:29]:
```suggestion
        <members>del_caseCollaborationComponent</members>
```

File: Delmar/manifest/package.xml [hunk 14]
```diff
@@ -1,31 +1,32 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <Package xmlns="http://soap.sforce.com/2006/04/metadata">
     <types>
-        <members>*</members>
+        <members>DEL_CommentsPageController</members>
```
Comment (pradeepchary) [inline Delmar/manifest/package.xml:4]:
```suggestion
        <members>DEL_CaseCollaborationController</members>
```

File: Delmar/manifest/package.xml [hunk 15]
```diff
@@ -1,31 +1,32 @@
...
+        <members>DEL_CommentsPageControllerTest</members>
```
Comment (pradeepchary) [inline Delmar/manifest/package.xml:5]:
```suggestion
        <members> DEL_CaseCollaborationControllerTest</members>
```

File: Delmar/force-app/main/default/classes/DEL_EmailMessagesTriggerHandler.cls (hunk 1 above)
Comment (pradeepchary) [inline Delmar/force-app/main/default/classes/DEL_EmailMessagesTriggerHandler.cls:22]:
```suggestion
        Pattern objPattern = Pattern.compile(strPattern);
```

**Comment by csankit0108** [comment]
@pradeepchary I made the suggested changes. Thanks

**Comment by pradeepchary** [comment]
I hope, the complete package is validated. Approving it

**Review by pradeepchary — APPROVED** [review]

**Pull request merged**